3scale-operator/tasks/main.yml

```diff
@@ -90,24 +130,55 @@
           apiVersion: apps.3scale.net/v1alpha1
           kind: APIcast
           metadata:
-            name: example-apicast-production
+            name: apicast-production
           spec:
+            deploymentEnvironment: production
+            configurationLoadMode: lazy
             adminPortalCredentialsRef:
               name: 3scaleportal
       register: result
-      wait: true
-      wait_condition:
-        ready: "True"
-      retries: 5
-      delay: 5
-      failed_when: result.resources[0] is not defined
+    - name: Check if APICast - Staging Deployment is Available
+      kubernetes.core.k8s_info:
+        api_version: apps/v1
+        kind: Deployment
+        name: apicast-apicast-staging
+        namespace: "{{ target_namespace }}"
+        wait: true
+        wait_condition:
+          status: "True"
+          type: "Available"
+        wait_sleep: 5
+        wait_timeout: 60
+      register: result
+    - name: Check if APICast - Production Deployment is Available
+      kubernetes.core.k8s_info:
+        api_version: apps/v1
+        kind: Deployment
+        name: apicast-apicast-production
+        namespace: "{{ target_namespace }}"
+        wait: true
+        wait_condition:
+          status: "True"
+          type: "Available"
+        wait_sleep: 5
+        wait_timeout: 60
+      register: result
+    - name: Check connectivity with the APICast - Staging service
+      ansible.builtin.uri:
+        url: http://apicast-apicast-staging:8080
```

Comment:
this will fail due to lack of cluster dns resolution. you could remove completely or test connectivity to `https://3scale-admin.3scale.{{ cluster_base_url }}.`